games-strategy/ufo-ai: Version bump to 2.5

quazgar opened the pull request:
Ebuild taken from bug #515660 and upgraded to EAPI 5.

Pull request as per https://bugs.gentoo.org/show_bug.cgi?id=515660

Builds and runs for me (amd64, proprietary nvidia drivers).

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+	sys-libs/zlib
+	editor? (
+		dev-libs/libxml2
+		virtual/jpeg
```
missing SLOT

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+DEPEND="!server? (
+		virtual/opengl
+		virtual/glu
+		media-libs/libsdl2
```
very likely missing USE flags here, double check https://wiki.gentoo.org/wiki/Project:Reviewers/Common_issues#Missing_or_incorrect_USE_flags_dependencies

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+# $Id$
+
+EAPI=5
+inherit eutils flag-o-matic games
```
games.eclass is deprecated by QA: https://wiki.gentoo.org/wiki/Project:Quality_Assurance/Meeting_Summaries#Games_team_policies_issue

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
+IUSE="debug server editor +client sse profile"
```
`server` and `client` are uncommon and non-standard USE flags for gentoo games, use `dedicated` instead like the original ebuild

quazgar replied:
Thanks for the comments so far.  I have one question here for clarification: would you suggest to drop the `client` USE flag completely --> by default build both server and client, and if `use dedicated` build just the server?

hasufell replied:
> by default build both server and client, and if use dedicated build just the server?

yes, that is expected behavior and also what the previous ebuild does (more or less...). If dedicated is not specified, just install everything.

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+
+src_unpack() {
+	unpack ${MY_P}-source.tar.bz2
+	cd "${S}"
```
missing `|| die`

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+src_unpack() {
+	unpack ${MY_P}-source.tar.bz2
+	cd "${S}"
+	pwd
```
?

quazgar replied:
Probably put here for debugging by the original author...  I am removing this line.

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+}
+
+src_install() {
+	newicon src/ports/linux/ufo.png ${PN}.png
```
use `newicon -s <size>` functions to install into modern locations and then gnome2-utils.eclass to update icon cache, see games-action/trosh for an example

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+	# install data
+	insinto "${GAMES_DATADIR}"/${PN/-}
+	doins -r base || die
+	rm -rf "${ED}/${GAMES_DATADIR}/${PN/-}/base/game.so"
```
why `-f` and why no `|| die`?

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+
+	# install data
+	insinto "${GAMES_DATADIR}"/${PN/-}
+	doins -r base || die
```
`doins` already has error handling, so no `|| die` needed here

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+pkg_setup() {
+	if use profile; then
+		ewarn "USE=\"profile\" is incompatible with the hardened profile's -pie flag."
+	fi
```
you walked into a trap and missed `games_pkg_setup` ;)
but as said... the eclass is deprecated anyway

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+		media-libs/sdl2-image[jpeg,png]
+		media-libs/sdl2-ttf
+		media-libs/sdl2-mixer
+		virtual/jpeg
```
missing SLOT

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+
+src_configure() {
+	# they are special and provide hand batched configure file
+	local myconf="
```
you should really make this a bash array

hasufell commented on games-strategy/ufo-ai/ufo-ai-2.5.ebuild:
```diff
@@ -0,0 +1,128 @@
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~x86"
```
drop `ppc`, since that is not keyworded for `media-libs/libsdl2`

mgorny commented:
@quazgar, are you still with us?

quazgar commented:
Yes, I just figured out, that apart from hasufell's comments (I fixed these issues) I found out that the install process is pretty borked. I am confident that I can get it running in the next days, I will get back here as soon as I'm done.

quazgar commented:
I think my ebuild works again now. I have one question regarding the use of the `games` eclass (said to be deprecated) and the implications for the install location: The current and previous ebuilds use the `GAMES_PREFIX` variable and others. Should I

1. just inherit games.eclass anyway, but only use the variables defined therein, not the functions?
2. define the install locations manually?
3. install the game not to `/usr/games/`, `/usr/share/games/` etc. but instead directly to `/usr/`?

hasufell commented:
don't use games.eclass.

mgorny commented:
@quazgar, expand a bit on what @hasufell said. Don't use the eclass or the paths defined there. Instead, use whatever upstream defaults to, unless it really doesn't make sense. So that'd be answer 3 in your case.

quazgar commented:
I changed the style according to your comments, thank you.  Additional changes:
- Removed the "editor" USE flag since I could not get uforadiant (the map editor) to work. This looks to me as if it could be fixed in the Makefiles.
- Similary the server program fails to find some files. I asked in the existing bug report upstream for updated map files, until then I think it's OK to print a warning.

Most importantly, the game is playable, hopefully this will motivate more skilled people to work on the remaining tasks.

monsieurp commented:
I cherry picked the 2 important commits and applied them. Please bear in mind for next time that if you want to sync your git repo with upstream, you ought to rebase the changes instead of merging them. Thank you. I'm closing this PR.

quazgar commented:
Hi @monsieurp, where did you applied the commits to? I could not find them in any branch of either the main repo or your clone. For everyone's convenience I squashed the commits and rebased the change on top of master in my fork.
